C2DEVEL-14695: return backup service support (#54)
C2Devel/terraform-provider-croccloud: website/docs/r/backup_selection.html.markdown

---
subcategory: "Backup"
layout: "aws"

> page_title: "AWS: aws_backup_selection"

Reviewer: ```suggestion
page_title: "aws_backup_selection"
```

Так для всех страничек

Author: Убрал

description: |-
  Manages selection conditions for AWS Backup plan resources.
---

# Resource: aws_backup_selection

> Manages selection conditions for AWS Backup plan resources.

Reviewer: ```suggestion
Manages selection conditions for backup plan resources.
```
\+ продублировать эту строку в тег `description` выше

Если имеется ввиду сервис, то с большой буквы. Если имеется ввиду ресурс (тут backup plan), то с маленькой.

Reviewer: Стараюсь использовать одинаковые формулировки в описании для ресурсов и датасорсов:
- для ресурса: `Manages a/an <resource>. For details about <service>, see the [user documentation](ссылка на облачную доку, если есть страница для сервиса).`
- для датасорса: `Provides information about a/an <resource>.`

В качестве примера смотри страницы для paas_\*, ec2_transit_gateway_\*

Author: Поправил

## Example Usage

### Selecting Backups By Resource

```terraform
resource "aws_backup_selection" "example" {

>   iam_role_arn = ""

Reviewer: Это неподдерживаемый параметр (по доке), можно убрать совсем

Author: Убрал, сделали computed полем

  name         = "tf_example_backup_selection"

>   plan_id      = aws_backup_plan.example.id

Reviewer: Если твой ресурс ссылается на другой ресурс, желательно приложить конфигурацию и для него тоже в пример, чтобы его можно было скопировать и запустить целиком.

Author: Добавил

>   resources = [

Reviewer: Кажется из этого списка мы поддерживаем только `aws_ebs_volume.example.arn`. Пример в доке должен быть рабочий + нужна конфигурация для указанных ресурсов

Author: Протестил еще с инстансом, но решил не пихать, чтобы не увеличивать и так объемный пример. Убрал все кроме ebs volume

    aws_db_instance.example.arn,
    aws_ebs_volume.example.arn,
    aws_efs_file_system.example.arn,
  ]
}
```

## Argument Reference

The following arguments are supported:

> * `name` - (Required) The display name of a resource selection document.

Reviewer: Для описания полей лучше брать версию из нашей документации, которая обычно более лаконична и однозначна. Также сюда добавляется информация о валидациях и дефолтных значениях для полей, возможных значениях если они есть. В качестве примера формулировок смотри страницы для paas_\*, ec2_transit_gateway_\*

Для редактируемых полей можно добавить `Editable` в начало. Например, `(Required, Editable)`

Author: Добавил

* `plan_id` - (Required) The backup plan ID to be associated with the selection of resources.

> * `iam_role_arn` - (Required) The ARN of the IAM role that AWS Backup uses to authenticate when restoring and backing up the target resource. See the [AWS Backup Developer Guide](https://docs.aws.amazon.com/aws-backup/latest/devguide/access-control.html#managed-policies) for additional information about using AWS managed policies or creating custom policies attached to the IAM role. _(Currently unsupported)_.

Reviewer: Изменить оформление неподдерживаемых полей

При переносе полей в неподдерживаемые атрибуты можно срезать часть описания до первого предложения + ссылки могут быть для нас не актуальны

Author: Поправил

* `selection_tag` - (Optional) Tag-based conditions used to specify a set of resources to assign to a backup plan. _(Currently unsupported)_.
* `condition` - (Optional) A list of conditions that you define to assign resources to your backup plans using tags. _(Currently unsupported)_.
* `resources` - (Optional) An array of strings that either contain Amazon Resource Names (ARNs) or match patterns of resources to assign to a backup plan.
* `not_resources` - (Optional) An array of strings that either contain Amazon Resource Names (ARNs) or match patterns of resources to exclude from a backup plan. _(Currently unsupported)_.

Tag conditions (`selection_tag`) support the following:

* `type` - (Required) An operation, such as `StringEquals`, that is applied to a key-value pair used to filter resources in a selection. _(Currently unsupported)_.
* `key` - (Required) The key in a key-value pair. _(Currently unsupported)_.
* `value` - (Required) The value in a key-value pair. _(Currently unsupported)_.

> ## Attributes Reference

Reviewer: ```suggestion
## Attribute Reference
```

Так для всех

Author: Поправил

In addition to all arguments above, the following attributes are exported:

> * `id` - Backup Selection identifier

Reviewer: ```suggestion
* `id` - The ID of the backup selection.
```

Author: Поправил, взял строку из доки

## Import

> Backup selection can be imported using the role plan_id and id separated by `|`.

Reviewer: ```suggestion
Backup selection can be imported using the `plan_id` and `id` separated by `|`.
```

Author: Убрал блок импорта в этом месте, не сработал

Author: Вернул

---
subcategory: "Backup"
layout: "aws"
page_title: "aws_backup_selection"
description: |-
  Manages selection conditions for backup plan resources.
---

# Resource: aws_backup_selection

Manages selection conditions for backup plan resources.

## Example Usage

### Selecting Backups By Resource

```terraform
resource "aws_backup_vault_default" "example" {
}

data "aws_availability_zones" "this" {
  state = "available"
}

resource "aws_ebs_volume" "example" {
  availability_zone = data.aws_availability_zones.this.names[0]

>   type              = "standard"

Reviewer: возьми тут `gp2` лучше

Author: Поправил

  size              = 8
}

resource "aws_backup_plan" "example" {
  name = "tf-backup-plan"

  rule {
    rule_name         = "tf-backup-rule"
    target_vault_name = aws_backup_vault_default.example.name
    schedule          = "cron(0 12 * * *)"
    start_window      = 60
    completion_window = 180

    lifecycle {
      delete_after = 30
    }
  }
}

resource "aws_backup_selection" "example" {
  name    = "tf-backup-selection"
  plan_id = aws_backup_plan.example.id

  resources = [
    aws_ebs_volume.example.arn,
  ]
}
```

## Argument Reference

The following arguments are supported:

> * `name` - (Required, Editable) The selection name. Must be at least `1` character and at most `50` characters in length. Must contain contain only alphanumeric, hyphen, underscore, and period characters.

Reviewer: ```suggestion
* `name` - (Required, Editable) The selection name. The value must be 1 to 50 characters long and must contain contain only alphanumeric, hyphen, underscore, and period characters.
```

Author: Поправил

* `plan_id` - (Required) The backup plan ID.
* `resources` - (Required, Editable) The list of Amazon Resource Names (ARNs) resources included into the backup plan.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The ID of the resource selection.

->  **Unsupported attributes**
These attributes are currently unsupported:

> * `condition` - A list of conditions that you define to assign resources to your backup plans using tags.

Reviewer: `Always empty` вероятно потерял + ниже

Author: Поправил

## Example Usage

### Selecting Backups By Resource

```terraform
resource "aws_backup_vault_default" "example" {}

data "aws_availability_zones" "this" {
  state = "available"
}

resource "aws_ebs_volume" "example" {
  availability_zone = data.aws_availability_zones.this.names[0]
  type              = "gp2"
  size              = 8
}

resource "aws_backup_plan" "example" {
  name = "tf-backup-plan"

  rule {
    rule_name         = "tf-backup-rule"
    target_vault_name = aws_backup_vault_default.example.name
    schedule          = "cron(0 12 * * *)"
    start_window      = 60
    completion_window = 180

    lifecycle {
      delete_after = 30
    }
  }

  rule {
    rule_name         = "tf-backup-rule2"
    target_vault_name = aws_backup_vault_default.example.name
    schedule          = "cron(0 23 * * *)"
    start_window      = 60
    completion_window = 180

    lifecycle {
      delete_after = 60
    }
  }
}

resource "aws_backup_selection" "example" {
  name    = "tf-backup-selection"
  plan_id = aws_backup_plan.example.id

  resources = [
    aws_ebs_volume.example.arn,
  ]
}
```

## Argument Reference

The following arguments are supported:

> * `name` - (Required, Editable) The selection name. The value must be `1` to `50` characters long and must contain only alphanumeric, hyphen, underscore, and period characters.

Reviewer 2: лучше: 
alphanumeric characters, hyphens, underscores, and periods.
и в других местах поменять

> * `plan_id` - (Required) The backup plan ID.

Reviewer 2: ID of ...

> * `resources` - (Required, Editable) The list of Amazon Resource Names (ARNs) resources included into the backup plan.

Reviewer 2: of the resources

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The ID of the resource selection.

->  **Unsupported attributes**
These attributes are currently unsupported:

* `condition` - A list of conditions that you define to assign resources to your backup plans using tags. Always empty.

> * `iam_role_arn` -  The Amazon Resource Name (ARN) of the role used for authentication when the resource is backed up (not implemented, added to ensure compatibility with AWS). Always `""`.

Reviewer 2: нужен текст в скобках?

> * `not_resources` - An array of strings that either contain Amazon Resource Names (ARNs) or match patterns of resources to exclude from a backup plan. Always empty.

Reviewer 2: массив же содержит? тогда
contains
matches

that contains either Amazon Resource Names (ARNs) or match patterns of resources to be excluded from a backup plan. Always empty.
